opensrp-chw-core/src/main/java/org/smartregister/chw/core/interactor/CoreChildProfileInteractor.java

```diff
@@ -502,6 +514,18 @@ public void createCarePlanEvent(@NotNull Context context, @NotNull String encode
         appExecutors.diskIO().execute(runnable);
     }
 
+    private void updateLocalStorage(String childBaseEntityId, String fieldName, String fieldValue) {
+        if (getChildAllCommonsRepository() != null) {
+            ContentValues values = new ContentValues();
+            values.put(fieldName, fieldValue);
+            getChildAllCommonsRepository().update("ec_child", values, childBaseEntityId);
+        }
```

Comment:
instead of using this custom logic whey don't we use ec_client_fields to extract this value. That is better since it also works for events synced from server

Reply:
I am using both, ec_client_fields took some time to process the event and update the local DB, we immediately need to update the local DB so that, accurate date of the care plan and care plan showed when the user has Thinkmd assessment.